src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -284,6 +285,11 @@ public void onBlockDropItems(BlockDropItemEvent e) {
         }
     }
 
+    @EventHandler
+    public void onBlockBreak(BlockBreakEvent e) {
+        Talisman.checkFor(e, SlimefunItems.TALISMAN_CAVEMAN);
```

Comment:
For redundancy but foremost also for performance reasons, there should probably be a check first whether the Player already has a `Haste` effect running.

Reply:
This check is done inside of the Talisman class, more specifically in the pass() method that gets called from within the checkFor() method